### PR TITLE
Fixed several issues found by static analysis

### PR DESCRIPTION
**Pull request by mshabunin**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/src/int8layers/eltwise_layer.cpp [hunk 1]
```diff
@@ -345,6 +345,8 @@ class EltwiseLayerInt8Impl CV_FINAL : public EltwiseLayerInt8
             int8_t* dstptr0 = dst->ptr<int8_t>();
             float* bufptr0 = buf->ptr<float>();
             int blockSize0 = 1 << 12;
+            CV_Assert(op == PROD && !zeropointsptr);
```
Comment (rogday) [inline modules/dnn/src/int8layers/eltwise_layer.cpp:348]: I think, you meant

```cpp
CV_Assert(op != PROD || !zeropointsptr);
```

**Review by rogday — CHANGES_REQUESTED** [review]
:+1:

File: modules/dnn/src/int8layers/eltwise_layer.cpp (hunk 1 above)
Comment (mshabunin) [reply modules/dnn/src/int8layers/eltwise_layer.cpp]: Sure, thank you!

File: modules/dnn/src/int8layers/eltwise_layer.cpp [hunk 2]
```diff
@@ -345,6 +345,8 @@ class EltwiseLayerInt8Impl CV_FINAL : public EltwiseLayerInt8
...
+            CV_Assert((op == PROD || op == SUM) && !coeffsptr);
```
[the remark this answers is not in the document]
Comment (mshabunin) [reply modules/dnn/src/int8layers/eltwise_layer.cpp]: Yes, thank you!

**Review by alalek — APPROVED** [review]

**Pull request merged**